### PR TITLE
sessions: fix proxy

### PR DESCRIPTION
**Pull request by prietopa**
Take proxy configuration from session if not specify in args of request method

### CODE REVIEW

**Comment by gabrielnoguerales** [comment]
Working great here
Good job @prietopa

**Comment by prietopa** [comment]
This pull request change order to merge proxy, verify, cert, etc.
It's an error to modify proxy before merge with session.proxy

File: requests/sessions.py [hunk 1]
```diff
@@ -693,6 +693,12 @@ def merge_environment_settings(self, url, proxies, stream, verify, cert):
 
         :rtype: dict
         """
+        # setting by configuration in Session.
+        proxies = merge_setting(proxies, self.proxies)
```
Comment (prietopa) [inline requests/sessions.py:697]: First, merge with session.proxy.
After, retrieve proxy information from os.environment.

**Comment by Kenith** [comment]
Hello,

Really appreciate for the fix.
But seems the bug fix not merged into master.
Would you please take a look at it?

Thank you,
Ken

**Comment by prietopa** [comment]
Hi Ken,

When I push merge button, in github web page, it said:
-"Merging is blocked", Merging can be performed automatically with 1 approving review"-
Can you approved?
or
I can do a new pull request in the master branch.

Thank you
prietopa

**Comment by Kenith** [comment]
Hello Prietopa,

Really appreciate for your time and your HOT fix.
I am not the contributor, seems that I don’t have access to approve your request.

@Gabronog Would you please take time to review Prietopa’s pull request.
Really appreciate for your help in advance.

Thank you,
Ken

**Comment by gabrielnoguerales** [comment]
> Hello Prietopa,
> 
> Really appreciate for your time and your HOT fix.
> I am not the contributor, seems that I don’t have access to approve your request.
> 
> @Gabronog Would you please take time to review Prietopa’s pull request.
> Really appreciate for your help in advance.
> 
> Thank you,
> Ken

Sure, it would be a pleasure :)
Thanks
Gabriel

**Comment by kennethreitz** [comment]
@nateprewitt what are you thoughts on this?

**Comment by kennethreitz** [comment]
let's just not change anything here. any delta in this area is potentially very dangerous for our users (unintended side-effects).

**Comment by prietopa** [comment]
@kennethreitz  What I wanted to say is that request is not taking the proxy configuration, changing these lines of code fixes it as it was being override.

**Pull request closed without merging**